_posts/2021-10-01-automated-testing-for-opensearch-releases.markdown

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  "Automated testing for OpenSearch releases"
+authors: 
+  - setiah
+date: 2021-10-01
+categories: 
+  - technical-post
+twittercard:
+  description: "This post details the automated testing process for OpenSearch releases"
+---
+
+# Automated testing for OpenSearch releases
+
+OpenSearch releases many distributions across multiple platform flavors as part of a new version release. These distributions are fundamentally of two kinds - the full distribution that includes all OpenSearch plugins packaged with OpenSearch core, and the min distribution which is just the core without any plugins. These release distributions go through a rigorous testing process across multiple teams, before they are signed off as release ready. This includes validating unit test coverage, integration testing across all components to validate the behavior, backward compatibility testing to ensure upgrade compatibility with previous versions, and stress testing to validate the performance metrics. Once these distributions are successfully tested and hardened, they are marked ready for the release.
+
+The rigorous release process provides good confidence in the quality of the release. However, for the most part it has been largely manual and non-standardized across plugins so far. Each plugin team used to validate their component by running tests on the distribution and provide their sign off. With dozens of OpenSearch plugins released as part of full distribution, the turn around time for testing was high. Also, lack of a continuous integration and testing process, would lead to finding new bugs at the time of release, which would further add to release times. 
```

Comment:
Lots of weasel words, generally shorten:

- largely, most part, so far are all related/synonyms, just say "The testing of the complete distribution was manual and varied across plugins".
- would lead -> lead
- would further add -> added